rust/lance/src/dataset/fragment.rs

```diff
@@ -2103,6 +2104,10 @@ impl FragmentReader {
                         })
                         .boxed()
                 })
+                .stream_in_span(tracing::debug_span!(
+                    "ReadBatchFutStream",
+                    id = self.fragment_id,
+                ))
```

Comment:
Hmm...I would think the underlying stream would already have a span attached to it.  Still, can't hurt.

Reply:
I didn't think I saw one, but I can remove if we find it's redundant.